reference/test/preconditioner/jacobi.cpp

```diff
@@ -494,4 +494,24 @@ TYPED_TEST(Jacobi, ScalarJacobiGeneratesOnDifferentPrecision)
 }
 
 
+TYPED_TEST(Jacobi, ScalarJacobiHandleZero)
+{
+    using value_type = typename TestFixture::value_type;
+    using Vec = typename TestFixture::Vec;
+    using Bj = typename TestFixture::Bj;
+    auto mtx = gko::share(
+        gko::initialize<Vec>({{0, 0, 0}, {0, 2, 0}, {0, 0, 0}}, this->exec));
+    auto b = gko::initialize<Vec>({1, 2, 3}, this->exec);
+    auto x = Vec::create(this->exec, gko::dim<2>(3, 1));
+    auto jacobi = this->scalar_j_factory->generate(mtx);
```

Comment:
nit: I would say that this belongs to the ACT part, so it should be in the second block, and the apply would be in the third block.

Reply:
It's mixed actually because it was also used to check the apply operation result.
I will move the generation to the ACT part but not move the apply

Reply:
I am wrong. I mix it with another pr. The apply is already tested in other test. here only checks the generation